src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -0,0 +1,74 @@
+package com.terraformation.backend.tracking
+
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.tracking.ObservationId
+import com.terraformation.backend.db.tracking.ObservationState
+import com.terraformation.backend.log.perClassLogger
+import com.terraformation.backend.log.withMDC
+import com.terraformation.backend.tracking.db.ObservationAlreadyStartedException
+import com.terraformation.backend.tracking.db.ObservationHasNoPlotsException
+import com.terraformation.backend.tracking.db.ObservationStore
+import com.terraformation.backend.tracking.db.PlantingSiteStore
+import com.terraformation.backend.tracking.model.PlantingSiteDepth
+import javax.inject.Named
+
+@Named
+class ObservationService(
+    private val observationStore: ObservationStore,
+    private val plantingSiteStore: PlantingSiteStore,
+) {
+  private val log = perClassLogger()
+
+  fun startObservation(observationId: ObservationId) {
+    requirePermissions { manageObservation(observationId) }
+
+    log.withMDC("observationId" to observationId) {
```

Comment:
Yes, that's correct. In dev environments it gets appended to the text of the log messages in the console. On Datadog in prod/staging, the name/value pair is recorded as metadata on the log message rather than as plain text.